internal/services/powertrade_service.go

```diff
@@ -0,0 +1,147 @@
+package services
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/goemon/internal/domain/entities"
+
+	websocket "github.com/goemon/internal/api/websockets/marketdata"
+	powertrade "github.com/goemon/internal/infra/powertrade"
+)
+
+type PowerTradeService struct {
+	client   *powertrade.PowerTradeClient
+	wsServer *websocket.WebSocketServer
+}
```

Comment:
```suggestion
type PowerTradeService struct {
	client   *powertrade.PowerTradeClient
	wsServer *websocket.WebSocketServer
	ctx      context.Context
	cancel   context.CancelFunc
}
```

use context for cancellation